### PR TITLE
Add support for EIP 1599

### PR DESCRIPTION
**Pull request by harshjv**

### CODE REVIEW

File: packages/bitcoin-node-wallet-provider/lib/BitcoinNodeWalletProvider.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export default class BitcoinNodeWalletProvider extends WalletProvider {
 
   async sendTransaction(options: SendOptions) {
     return options.fee
-      ? this.withTxFee(async () => this._sendTransaction(options), options.fee)
+      ? this.withTxFee(async () => this._sendTransaction(options), options.fee as number)
```
Comment (monokh) [inline packages/bitcoin-node-wallet-provider/lib/BitcoinNodeWalletProvider.ts:76]: Hmm weird, why do we need to cast `as number` everywhere? It looks like the places you are passing this, the type is already `number` and the destination type is `number`

File: packages/ethereum-eip1559-fee-provider/lib/EthereumEIP1559FeeProvider.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  _slowMultiplier: number
+  _averageMultiplier: number
+  _fastMultiplier: number
+  _maxFeePerGasThreshold: number
+
+  constructor(options: { uri: string; username?: string; password?: string }, opts: FeeOptions = {}) {
+    super(options.uri, options.username, options.password)
+
+    const { slowMultiplier = 1, averageMultiplier = 1.25, fastMultiplier = 1.4, maxFeePerGasThreshold = 1000 } = opts
+    this._slowMultiplier = slowMultiplier
```
Comment (monokh) [inline packages/ethereum-eip1559-fee-provider/lib/EthereumEIP1559FeeProvider.ts]: are the multipliers needed anymore?
I mean i thought the EIP1559 logic itself can give good recommendations
the main use case of this multiplier was that the estimation come from node for legacy transaction was not very accurate

File: packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.ts [hunk 3]
```diff
@@ -68,47 +72,87 @@ export default class EthereumLedgerProvider extends LedgerProvider<HwAppEthereum
...
     const txOptions: ethereum.UnsignedTransaction = {
       from,
       to: options.to ? addressToString(options.to) : (options.to as string),
       value: options.value,
       data: options.data,
-      gasPrice,
       nonce
     }
 
+    if (options.fee) {
```
Comment (monokh) [inline packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.ts]: Wow there is a lot of duplication with JsProvider 
Definitely will be good to have a EthereumWalletProvider base class
We can do separately/later

**Review by monokh — APPROVED** [review]

File: packages/ethereum-eip1559-fee-provider/lib/EthereumEIP1559FeeProvider.ts (hunk 2 above)
Comment (harshjv) [reply packages/ethereum-eip1559-fee-provider/lib/EthereumEIP1559FeeProvider.ts]: Started using https://github.com/rainbow-me/fee-suggestions and it seems to be handling this well with surge in fees.

**Pull request merged**